packages/headless/src/features/search-parameters/search-parameter-actions.ts

```diff
@@ -1,33 +1,73 @@
 import {createAction} from '@reduxjs/toolkit';
-import {SearchRequest} from '../../api/search/search/search-request';
 import {validatePayload} from '../../utils/validate-payload';
 import {DateRangeRequest} from '../facets/range-facets/date-facet-set/interfaces/request';
 import {NumericRangeRequest} from '../facets/range-facets/numeric-facet-set/interfaces/request';
 import {searchParametersDefinition} from './search-parameter-schema';
 
-type FacetParameters = {
-  f: Record<string, string[]>;
-  cf: Record<string, string[]>;
-  nf: Record<string, NumericRangeRequest[]>;
-  df: Record<string, DateRangeRequest[]>;
-};
-
-export type SearchParameters = Omit<
-  SearchRequest,
-  | 'organizationId'
-  | 'accessToken'
-  | 'visitorId'
-  | 'url'
-  | 'enableDidYouMean'
-  | 'fieldsToInclude'
-  | 'facetOptions'
-  | 'facets'
-  | 'searchHub'
-  | 'pipeline'
-  | 'context'
-  | 'locale'
-> &
-  Partial<FacetParameters>;
+/**
+ * The parameters affecting the search response.
+ */
+export interface SearchParameters {
+  /**
+   * The advanced query expression.
+   */
+  aq?: string;
+
+  /**
+   * A record of the category facets, where the key is the facet id, and values are the selected path segements.
```

Comment:
`are the selected path segments` is this supposed to mean the path to the selected value or something else? It's a little unclear to me.

Reply:
It is the path to the select value. The path should be a `string[]` where each element is a part of the path.